src/frontend/xgboost_json.cc

```diff
@@ -25,45 +43,363 @@ namespace frontend {
+bool LearnerParamHandler::String(const char *str,
+                                 std::size_t length,
+                                 bool copy) {
+  return (assign_value("base_score", strtof(str, nullptr),
+                       m_output.param.global_bias) ||
+          assign_value("num_class", static_cast<int>(std::max(atoi(str), 1)),
+                       m_output.num_output_group) ||
+          assign_value("num_feature", atoi(str), m_output.num_feature));
+}
+
+/******************************************************************************
+ * LearnerHandler
+ * ***************************************************************************/
+bool LearnerHandler::StartArray() {
+  return push_key_handler<IgnoreHandler>("eval_metrics"); }
+
+bool LearnerHandler::StartObject() {
+  // "attributes" key is not documented in schema
+  return (push_key_handler<LearnerParamHandler, treelite::Model>(
+              "learner_model_param", m_output) ||
+          push_key_handler<GradientBoosterHandler, treelite::Model>(
+              "gradient_booster", m_output) ||
+          push_key_handler<ObjectiveHandler, std::string>("objective",
+                                                          objective) ||
+          push_key_handler<IgnoreHandler>("attributes"));
+}
+
+bool LearnerHandler::EndObject(std::size_t memberCount) {
+  if (objective == "multi:softmax") {
+    std::strncpy(m_output.param.pred_transform, "max_index",
+                 sizeof(m_output.param.pred_transform));
+  } else if (objective == "multi:softprob") {
+    std::strncpy(m_output.param.pred_transform, "softmax",
+                 sizeof(m_output.param.pred_transform));
+  } else if (objective == "reg:logistic" || objective == "binary:logistic") {
+    std::strncpy(m_output.param.pred_transform, "sigmoid",
+                 sizeof(m_output.param.pred_transform));
+  } else if (objective == "count:poisson" || objective == "reg:gamma" ||
+             objective == "reg:tweedie") {
+    std::strncpy(m_output.param.pred_transform, "exponential",
+                 sizeof(m_output.param.pred_transform));
+  } else {
+    std::strncpy(m_output.param.pred_transform, "identity",
+                 sizeof(m_output.param.pred_transform));
+  }
+  return pop_handler();
+}
+
+/******************************************************************************
+ * XGBoostModelHandler
+ * ***************************************************************************/
+bool XGBoostModelHandler::StartArray() {
+  return push_key_handler<ArrayHandler<unsigned>, std::vector<unsigned>>(
+      "version", version);
+}
+
+bool XGBoostModelHandler::StartObject() {
+  return push_key_handler<LearnerHandler, treelite::Model>("learner",
+                                                           m_output);
+}
+
+bool XGBoostModelHandler::EndObject(std::size_t memberCount) {
+  if (memberCount != 2) {
+    return false;
+  }
+  m_output.random_forest_flag = false;
+  if (version[0] >= 1) {
```

Comment:
```suggestion
  // Before XGBoost 1.0.0, the global bias saved in model is a transformed value.  After
  // 1.0 it's the original value provided by user.
  if (version[0] >= 1) {
```
Let's add a comment to describe why we need to transform the margin here.